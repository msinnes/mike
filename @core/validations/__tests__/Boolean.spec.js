const BooleanValidation = require('../Boolean');
const Validation = require('../classes/Validation');

describe('Boolean', () => {
  it('should be a function', () => {
    expect(BooleanValidation).toBeDefined();
    expect(BooleanValidation).toBeInstanceOf(Function);
  });

  it('should take an optional message argument', () => {
    expect(() => {
      BooleanValidation();
    }).not.toThrow();

    expect(() => {
      BooleanValidation('Boolean');
    }).not.toThrow();

    expect(() => {
      BooleanValidation(() => {});
    }).not.toThrow();

    expect(() => {
      BooleanValidation(1);
    }).toThrowErrorMatchingSnapshot();
  });

  it('should take an optional config argument', () => {
    expect(() => {
      BooleanValidation('string');
    }).not.toThrow();

    expect(() => {
      BooleanValidation('string', {});
    }).not.toThrow();

    expect(() => {
      BooleanValidation('string', 1);
    }).toThrowErrorMatchingSnapshot();
  });

  it('should return an instance of Validation', () => {
    expect(BooleanValidation()).toBeInstanceOf(Validation);
    expect(BooleanValidation().validate).toBeDefined();
  });

  it('should test if a value is a boolean', () => {
    const validation = BooleanValidation();
    const firstResult = validation.validate(true);
    expect(firstResult.valid).toBeDefined();
    expect(firstResult.valid).toBe(true);

    const secondResult = validation.validate();
    expect(secondResult.invalid).toBeDefined();
    expect(secondResult.invalid).toBe(true);
  });
});