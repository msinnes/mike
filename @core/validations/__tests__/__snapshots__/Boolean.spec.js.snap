// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Boolean should take an optional config argument 1`] = `"InternalError: the configuration is invalid, allowed keys are allowEmpty, throwOnInvalid, and type"`;

exports[`Boolean should take an optional message argument 1`] = `"InternalError: message must be a string or a function"`;
